.gitignore

```diff
@@ -17,3 +17,20 @@ wheels/
 
 #Environment Variables
 .env
+
+# Frontend: Node modules
+node_modules/
+
+# Frontend: Build output
+build/
+dist/
+
+# Frontend: Dependency directories
+/.pnp
+.pnp.js
+
+# Frontend: Logs
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+pnpm-debug.log*
```

Comment:
should these be mace more specific to the `frontend` directory to avoid conflict with other things, e.g. it's plausible that a `build` or `dist` directory might live somewhere else in the repo also (though I suppose it too would likely benefit from being ignored, maybe)

Reply:
Totally valid concern. will update